src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -266,7 +266,7 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr curlPtr)
                 catch (Exception exc)
                 {
                     EventSourceTrace("Unexpected exception: {0}", exc, easy: easy);
-                    easy.FailRequest(exc);
+                    easy.FailRequest(CreateHttpRequestException(exc));
```

Comment:
After further investigation on .NET Framework behavior, it seems that the exception wrapping is as follows:

* HttpWebRequest throws a WebException and inside that is the exception bubbling from the user-provided callback.
* HttpClient throws an HttpRequestException. Inside that is a WebException.  That is behavior on .NET Framework since HttpCLient uses HttpWebRequest internally.  And inside that WebException is the exception bubbling from the user-provided callback.

So, I would think we want the behavior in .NET Core to perhaps have HttpRequestException as outer and then inside that is a handler exception (WInHttpException or CurlException) and inside that is the exception from the user-provided callback.

Developers can find the innermost exception by using 1Exception.GetBaseException()'